labs/celery.py

```diff
@@ -0,0 +1,269 @@
+from celery import Celery, chain
+from celery.signals import task_failure
+from kombu import Queue
+from redbeat import RedBeatSchedulerEntry, schedulers
+import redis
+from labs.config import settings
+import logging
+import json
+
+from labs.database.embeddings import find_similar_embeddings
+from labs.database.vectorize import vectorize_to_database
+from labs.repo import call_agent_to_apply_code_changes, clone_repository
+from labs.llm import get_llm_response, get_prompt, prepare_context
+from labs.run import commit_changes, create_branch, create_pull_request, get_issue
+
+
+logger = logging.getLogger(__name__)
+
+CELERY_QUEUE_PREFIX = "code-monkey"
+DEFAULT_QUEUE_NAME = CELERY_QUEUE_PREFIX
+LOW_PRIORITY_QUEUE_NAME = f"{CELERY_QUEUE_PREFIX}-low"
+HIGH_PRIORITY_QUEUE_NAME = f"{CELERY_QUEUE_PREFIX}-high"
+
+redis_client = redis.StrictRedis(host="redis", port=6379, db=0, decode_responses=True)
```

Comment:
these values must be retrieved from env variables

Reply:
fixed